src/cargo/util/toml/mod.rs

```diff
@@ -1608,161 +1643,36 @@ impl InheritableFields {
         self.ws_root = ws_root;
     }
 
-    pub fn dependencies(&self) -> CargoResult<BTreeMap<String, TomlDependency>> {
-        self.dependencies.clone().map_or(
-            Err(anyhow!("`workspace.dependencies` was not defined")),
-            |d| Ok(d),
-        )
-    }
-
-    pub fn lints(&self) -> CargoResult<TomlLints> {
-        self.lints
-            .clone()
-            .map_or(Err(anyhow!("`workspace.lints` was not defined")), |d| Ok(d))
-    }
-
+    /// Gets a workspace dependency with the `name`.
     pub fn get_dependency(&self, name: &str, package_root: &Path) -> CargoResult<TomlDependency> {
-        self.dependencies.clone().map_or(
-            Err(anyhow!("`workspace.dependencies` was not defined")),
-            |deps| {
-                deps.get(name).map_or(
-                    Err(anyhow!(
-                        "`dependency.{}` was not found in `workspace.dependencies`",
-                        name
-                    )),
-                    |dep| {
-                        let mut dep = dep.clone();
-                        if let TomlDependency::Detailed(detailed) = &mut dep {
-                            detailed.resolve_path(name, self.ws_root(), package_root)?
-                        }
-                        Ok(dep)
-                    },
-                )
-            },
-        )
-    }
-
-    pub fn version(&self) -> CargoResult<semver::Version> {
-        self.version.clone().map_or(
-            Err(anyhow!("`workspace.package.version` was not defined")),
-            |d| Ok(d),
-        )
-    }
-
-    pub fn authors(&self) -> CargoResult<Vec<String>> {
-        self.authors.clone().map_or(
-            Err(anyhow!("`workspace.package.authors` was not defined")),
-            |d| Ok(d),
-        )
-    }
-
-    pub fn description(&self) -> CargoResult<String> {
-        self.description.clone().map_or(
-            Err(anyhow!("`workspace.package.description` was not defined")),
-            |d| Ok(d),
-        )
-    }
-
-    pub fn homepage(&self) -> CargoResult<String> {
-        self.homepage.clone().map_or(
-            Err(anyhow!("`workspace.package.homepage` was not defined")),
-            |d| Ok(d),
-        )
-    }
-
-    pub fn documentation(&self) -> CargoResult<String> {
-        self.documentation.clone().map_or(
-            Err(anyhow!("`workspace.package.documentation` was not defined")),
-            |d| Ok(d),
-        )
+        let Some(deps) = &self.dependencies else {
```

Comment:
Nit: It feels odd to have the getters split up by the `update_*` functions; it might be better if all the getters were beside one another. 

Note: [`ws_root`](https://github.com/rust-lang/cargo/blob/5febbe5587b74108165f748e79a4f8badbdf5e0e/src/cargo/util/toml/mod.rs#L1768) could also be moved, but it is not the exact same as the other getters. I could go either way so I will leave the decision up to you.